tensorflow/core/kernels/maxpooling_op.cc

```diff
@@ -1311,7 +1334,12 @@ template <typename T>
 struct LaunchMaxPoolingGradWithArgmax<Eigen::GpuDevice, T> {
   static void launch(OpKernelContext* context, const PoolParameters& params,
                      const Tensor& grad_in, const Tensor& argmax,
-                     Tensor* grad_out) {
+                     Tensor* grad_out, const bool include_batch_in_index) {
+    // TODO: support include_batch_in_index=true for gpu kernel.
+    OP_REQUIRES(context, include_batch_in_index == false,
```

Comment:
We can add a check for this attr at kernel registration time (just like how we restrict kernels for certain dtypes)